src/platform-implementation-js/dom-driver/gmail/views/gmail-route-view/gmail-route-view.ts

```diff
@@ -293,15 +295,30 @@ class GmailRouteView {
   }
 
   _setupSubViews() {
-    asap(() => {
+    asap(async () => {
       if (!this._eventStream) return;
 
+      await this.#waitForMainElementSafe();
+
       this.#monitorRowListElements();
       this._setupContentAndSidebarView();
       this._setupScrollStream();
     });
   }
 
+  async #waitForMainElementSafe() {
+    try {
+      // role=main attribute is not set while page in a loading state
+      await waitFor(() => document.querySelector('[role=main]'), 15_000);
+    } catch {
+      this._driver
+        .getLogger()
+        .error(new Error('[role=main] element not found'), {
```

Comment:
We might use `SelectorError` here instead of `Error`.

Reply:
changed